lib/fastly/client.rb

```diff
@@ -102,6 +102,11 @@ def purge(url, params = {})
       extras = params.delete(:headers) || {}
       uri    = URI.parse(url)
       http   = Net::HTTP.new(uri.host, uri.port)
+
+      if uri.is_a? URI::HTTPS
```

Comment:
Please use `if uri.scheme.downcase == 'https'` instead of the class check because it is possible to have a URI::HTTP that has an 'https' scheme.